src/main/java/com/conveyal/datatools/manager/models/Deployment.java

```diff
@@ -385,6 +374,35 @@ public void dump (File output, boolean includeManifest, boolean includeOsm, bool
         out.close();
     }
 
+    /** Generate build config for deployment as byte array (for writing to file output stream). */
+    public byte[] generateBuildConfig() throws JsonProcessingException {
+        Project project = this.parentProject();
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.setSerializationInclusion(Include.NON_NULL);
+        return customBuildConfig != null
+            ? customBuildConfig.getBytes(StandardCharsets.UTF_8)
+            : project.buildConfig != null
+            ? mapper.writer().writeValueAsBytes(project.buildConfig)
+            : null;
+    }
+
+    /** Generate router config for deployment as byte array (for writing to file output stream). */
+    public byte[] generateRouterConfig() {
+        Project project = this.parentProject();
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.setSerializationInclusion(Include.NON_NULL);
+        try {
+            return customRouterConfig != null
+                ? customRouterConfig.getBytes(StandardCharsets.UTF_8)
+                : project.routerConfig != null
+                ? mapper.writer().writeValueAsBytes(project.routerConfig)
+                : null;
+        } catch (JsonProcessingException e) {
+            LOG.error("Router config contains malformed JSON", e);
+            return null;
+        }
+    }
```

Comment:
Lots of common code here. Seems like another method could be created with just the ObejctMapper writing stuff to bytes part.